include/modem/lte_lc.h

```diff
@@ -42,38 +52,111 @@ enum lte_lc_system_mode {
 };
 
 /* NOTE: enum lte_lc_func_mode maps directly to the functional mode
- *	 as returned by the AT command "AT+CFUN?".
+ *	 as returned by the AT command "AT+CFUN?" as specified in
+ *	 "nRF91 AT Commands - Command Reference Guide v1.1"
  */
 enum lte_lc_func_mode {
 	LTE_LC_FUNC_MODE_POWER_OFF		= 0,
 	LTE_LC_FUNC_MODE_NORMAL			= 1,
 	LTE_LC_FUNC_MODE_OFFLINE		= 4,
-	LTE_LC_FUNC_MODE_OFFLINE_UICC_ON	= 44
+	LTE_LC_FUNC_MODE_DEACTIVATE_LTE		= 20,
+	LTE_LC_FUNC_MODE_ACTIVATE_LTE		= 21,
+	LTE_LC_FUNC_MODE_DEACTIVATE_GNSS	= 30,
+	LTE_LC_FUNC_MODE_ACTIVATE_GNSS		= 31,
+	LTE_LC_FUNC_MODE_OFFLINE_UICC_ON	= 44,
+};
+
+enum lte_lc_evt_type {
+	LTE_LC_EVT_NW_REG_STATUS,
+	LTE_LC_EVT_PSM_UPDATE,
+	LTE_LC_EVT_EDRX_UPDATE,
+	LTE_LC_EVT_RRC_UPDATE,
+	LTE_LC_EVT_CELL_UPDATE,
+};
+
+enum lte_lc_rrc_mode {
+	LTE_LC_RRC_MODE_IDLE,
+	LTE_LC_RRC_MODE_CONNECTED,
+};
+
+struct lte_lc_psm_cfg {
+	int tau;		/* Periodic Tracking Area Update interval */
+	int active_time;	/* Active-time (time from RRC idle to PSM) */
+};
+
+struct lte_lc_edrx_cfg {
+	float edrx;	/* eDRX interval value [s] */
+	float ptw;	/* Paging time window [s] */
+};
+
+struct lte_lc_cell {
+	u32_t id;	/* E-UTRAN cell ID */
+	u32_t tac;	/* Tracking Area Code */
+};
+
+struct lte_lc_evt {
+	enum lte_lc_evt_type type;
+	union {
+		enum lte_lc_nw_reg_status nw_reg_status;
+		enum lte_lc_rrc_mode rrc_mode;
+		struct lte_lc_psm_cfg psm_cfg;
+		struct lte_lc_edrx_cfg edrx_cfg;
+		struct lte_lc_cell cell;
+	};
 };
 
-/** @brief Function for initializing
- * the modem.  NOTE: a follow-up call to lte_lc_connect()
- * must be made.
+typedef void(*lte_lc_evt_handler_t)(const struct lte_lc_evt *const evt);
+
+/** @brief Register event handler for LTE events.
+ *
+ *  @param handler Event handler. Handler is de-registered if parameter is
+ *		   NULL.
+ */
+void lte_lc_register_handler(lte_lc_evt_handler_t handler);
+
+/** @brief Function for initializing the modem.
```

Comment:
Reading at the documentation once more here.

This has nothing to do with Modem initialization. It only initializes the module but does not send any commands to the Modem.